src/GraphQL.MicrosoftDI.Tests/ConnectionBuilderExtensionTests.cs

```diff
@@ -39,7 +39,7 @@ public void WithScope0()
                 .Resolve()
                 .WithScope()
                 .Resolve(context => "hello");
-            field.Resolver.Resolve(_scopedContext).ShouldBe("hello");
+            field.Resolver.ResolveAsync(_scopedContext).Result.ShouldBe("hello");
```

Comment:
If resolver body is async of course.

Reply:
I was thinking of changing all those to run asynchronously.  Some of the unchanged code also is running synchronously. 
 But the bigger problem is that in GraphQL.Tests, every time we use QueryTestBase, it uses `GetAwaiter().GetResult()` for every validation call.  So while here I only need to change two files and it will all be async, in the main project we would have a lot of changes to make.

Reply:
At the moment, I would like to merge this code as-is, without further complicating this PR.  I'm trying to keep this PR to a minimum of changes -- just get ValueTask in place first before continuing work.

Reply:
OK.

Reply:
I even can't resolve conversations that I started.